ext/drstatecmp/drstatecmp.c

```diff
@@ -483,7 +483,8 @@ drstatecmp_check_simd_value
 #elif defined(AARCHXX)
     (void *tag, dr_simd_t *value, dr_simd_t *expected)
 {
-    if (memcmp(value, expected, sizeof(dr_simd_t)))
+    size_t vl = proc_has_feature(FEATURE_SVE) ? proc_get_sve_vector_length_bytes() : 16;
```

Comment:
Should we rename proc_get_sve_vector_length_bytes() to proc_get_vector_length_bytes() so callers don't have to check whether SVE is enabled and can do a simple single call?

Reply:
Do you mean have the `proc_has_feature(FEATURE_SVE)` check in `proc_get_vector_length_bytes()` and return either the scalable VL or 16?

I tried that as it's cleaner but the problem is it fails when decoding SVE instructions on non SVE h/w. Trying to set a universal (SIMD or SVE) vector length on startup in `cpu_info.vector_length_bytes` fails for the same reason. Bear in mind we want to be able to run **_one_** base v8 binary on SVE and non-SVE h/w.

Reply:
> Do you mean have the `proc_has_feature(FEATURE_SVE)` check in `proc_get_vector_length_bytes()` and return either the scalable VL or 16?

Yes, exactly.

> I tried that as it's cleaner but the problem is it fails when decoding SVE instructions on non SVE h/w. Trying to set a universal (SIMD or SVE) vector length on startup in `cpu_info.vector_length_bytes` fails for the same reason. Bear in mind we want to be able to run **_one_** base v8 binary on SVE and non-SVE h/w.

Not sure I understand: won't we only try to get the SVE vector length by executing an instruction if we know we're on SVE hardware?  Or is that not what you mean?  

(Note: here we have vector_length spelled out in the API routine proc_get_sve_vector_length_bytes; xref comment above about dr_set_sve_vl.)

Reply:
Done. `cpu_info.sve_vector_length_bytes` set in one place (`get_processor_specific_info()`) so `proc_get_vector_length_bytes()` returns the correct length for SVE and non-SVE.